src/game/card.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+#include <deque>
+#include <set>
+#include <algorithm>
+#include <random>
+
+#include <uno/uno.pb.h>
+
+namespace SailGame { namespace Game {
+
+using ::Uno::CardColor;
+using ::Uno::CardText;
+
+struct Card;
+
+using InitHandcardsT = std::array<Card, 7>;
+
+struct Card {
+    CardColor mColor;
+    CardText mText;
+
+    Card() = default;
+    Card(CardColor color, CardText text) : mColor(color), mText(text) {}
+    Card(const ::Uno::Card &card) : mColor(card.color()), mText(card.text()) {}
+
+    Card(const Card &) = default;
+    Card &operator=(const Card &) = default;
+
+    ::Uno::Card ConvertToGrpcCard() const {
+        ::Uno::Card card;
+        card.set_color(mColor);
+        card.set_text(mText);
+        return card;
+    }
+
+    bool operator<(const Card &rhs) const {
+        return (mColor < rhs.mColor) || 
+            (mColor == rhs.mColor && mText < rhs.mText);
+    }
+    
+    bool operator==(const Card &rhs) const {
+        return (mColor == rhs.mColor) && (mText == rhs.mText);
+    }
+
+    bool operator!=(const Card &card) const {
+        return !(*this == card);
+    }
+};
+
+struct CardSet {
+    const static std::set<CardColor> NonWildColors;
+    const static std::set<CardText> NonWildTexts;
+    const static std::set<CardText> DrawTexts;
+};
+
+class Handcards {
+public:
+    Handcards() = default;
+
+    void Draw(Card card);
```

Comment:
Card& card?

also, if you don't change card,  const Card&?

Reply:
doesn't make much sense, `Card` contains only color and text, which are both enumration whose underlying type is `int`. briefly speaking, `sizeof(Card)` is 8, pass-by-value is ok